src/bin/stratis_uuids_to_names.rs

```diff
@@ -0,0 +1,201 @@
+        StratisUdevError::new(e)
+    }
+}
+
+impl Debug for StratisUdevError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Could not convert Stratis UUIDs to names.")?;
+        if let Some(ref msg) = self.0 {
+            write!(f, " {}", msg)
+        } else {
+            Ok(())
+        }
+    }
+}
+
+type GMORet =
+    HashMap<Path<'static>, HashMap<String, HashMap<String, Variant<Box<dyn RefArg + 'static>>>>>;
+
+fn get_managed_objects() -> Result<GMORet, StratisUdevError> {
+    let connection = Connection::new_system()?;
+    let proxy = connection.with_proxy(STRATIS_BUS_NAME, STRATIS_MANAGER_OBJECT, *TIMEOUT);
+    Ok(proxy
+        .method_call(DBUS_OM_IFACE, "GetManagedObjects", ())
+        .map(|r: (GMORet,)| r.0)?)
+}
+
+fn udev_name_to_uuids(dm_name: &str) -> Result<Option<(Uuid, Uuid)>, StratisUdevError> {
+    let regex = Regex::new("stratis-1-(?P<pool>[0-9a-f]{32})-thin-fs-(?P<fs>[0-9a-f]{32})")?;
+    let captures = match regex.captures(dm_name) {
+        Some(cap) => cap,
+        None => return Ok(None),
+    };
+    let pool_uuid = captures
+        .name("pool")
+        .map(|pu| Uuid::parse_str(pu.as_str()).expect("Format validated by regex"));
+    let fs_uuid = captures
+        .name("fs")
+        .map(|pu| Uuid::parse_str(pu.as_str()).expect("Format validated by regex"));
+    Ok(pool_uuid.and_then(|pu| fs_uuid.map(|fu| (pu, fu))))
+}
+
+fn uuid_to_stratis_name(
+    managed_objects: &GMORet,
+    iface_name: &'static str,
+    uuid: Uuid,
+) -> Result<Option<String>, StratisUdevError> {
+    let mut names: Vec<_> = managed_objects
+        .values()
+        .filter_map(|map| {
+            if map.contains_key(iface_name)
+                && map
+                    .get(iface_name)
+                    .and_then(|submap| submap.get("Uuid").and_then(|uuid| uuid.as_str()))
+                    == Some(&uuid.to_simple_ref().to_string())
+            {
+                map.get(iface_name).and_then(|submap| {
+                    submap
+                        .get("Name")
+                        .and_then(|name| name.as_str().map(|n| n.to_string()))
+                })
+            } else {
+                None
+            }
+        })
+        .collect();
+
+    if names.len() > 1 {
+        Err(StratisUdevError::new(format!(
+            "More than one device has the UUID {}",
+            uuid.to_simple_ref()
+        )))
+    } else {
+        Ok(names.pop())
+    }
+}
+
+fn main_report_error() -> Result<Option<(String, String)>, StratisUdevError> {
+    let mut args = args();
+    let dm_name = match args.nth(1) {
+        Some(dm_name) => dm_name,
+        None => {
+            return Err(StratisUdevError::new(
+                "Thinly provisioned filesystem devicemapper name required as argument.",
+            ));
+        }
+    };
+    match args.next().as_deref() {
+        Some(action) => {
+            if action == "remove" {
```

Comment:
I get this from udevadm:

```
$ udevadm test --action=help
add
remove
change
move
online
offline
bind
unbind
```

I _think_ we'll probably want to exclude them all except for ```add``` and ```change```.